grunt/publish-modules.js

```diff
@@ -0,0 +1,94 @@
+module.exports = function(grunt) {
+  var fs = require('fs');
+  var path = require('path');
+  var spawn = require('child_process').spawn;
+
+  function npmPublish(cwd) {
+    return new Promise(function(resolve, reject) {
+      if (grunt.option('dry-run')) {
+        grunt.log.write('spawn npm publish {cwd: ' + cwd +'}\n');
+        resolve();
+      }
+      else {
+        var child = spawn('npm', ['publish'], {
+          stdio: 'inherit',
+          cwd: path.resolve(cwd || ''),
+        });
+        child.on('error', reject);
+        child.on('exit', resolve);
+      }
+    });
+  }
+
+  function promisify(fn, ctx) {
+    return function() {
+      var args = [].slice.call(arguments);
+      return new Promise(function(resolve, reject) {
+        args.push(function(error, result) {
+          if (error) {
+            reject(error);
+          }
+          else if (arguments.length === 2) {
+            resolve(result);
+          }
+          else {
+            resolve([].slice.call(arguments, 1));
+          }
+        });
+        fn.apply(ctx, args);
+      });
+    };
+  }
+
+  grunt.registerTask('publish-modules', function() {
```

Comment:
segregated. It shouldn't be used frequently. I almost want to add an interactive check to make it's what the user wants to do. Very few people will use this, we just want to remove the error.

I guess the other part is I'm not sure what I'd name the `npm run` script considering `npm publish`. I suspect it'd be confusing. Though I'm super open to suggestions. I'd like to do something other than grunt publish too. How does `npm run deploy` or `grunt deploy` sound to you?